### PR TITLE
Added support for promises

### PR DESCRIPTION
**Pull request by mcollina**
I find myself recommending async/await more and more for writing business logic, and I think this module would benefit from providing native support for promises.

### CODE REVIEW

**Comment by emilbayes** [comment]
Looks good! I'm pretty keen on landing this soon: https://github.com/emilbayes/secure-password/pull/5/files

How will that work with promises?

The idea is basically to queue the operations and return a function that can maybe be used to cancel if it hasn't started the expensive hashing yet

**Comment by mcollina** [comment]
We have two options: a) a user have to use callbacks to leverage that
feature b) the promises have a cancel() method attached which is the
function returned.
We might just go for option a).
Il giorno dom 2 set 2018 alle 18:32 Emil Bay <notifications@github.com> ha
scritto:

**Comment by mcollina** [comment]
all of that will remain transparent, the promise logic should stay in the
.verify and .hash methods.
Il giorno dom 2 set 2018 alle 19:24 Emil Bay <notifications@github.com> ha
scritto:

File: index.js [hunk 1]
```diff
@@ -53,6 +53,20 @@ SecurePassword.prototype.hashSync = function (passwordBuf) {
...
+      this.hash(passwordBuf, function (err, hashBuf) {
+        if (err) {
+          reject(err)
+          return
+        }
+
+        resolve(hashBuf)
+      })
+    })
+  }
```
Comment (vweevers) [inline index.js]: [catering](https://github.com/vweevers/catering) might interest you, wrote it 2 days ago. Would reduce the above to:

```js
const catering = require('catering').attach

SecurePassword.prototype.hash = function (passwordBuf, cb) {
  cb = catering(cb)
  // ..
  return cb.promise
}
```

**Comment by emilbayes** [comment]
Ok, I will allow it! Coming out in the next minor release

**Comment by emilbayes** [comment]
Released as `3.1.0`

**Pull request merged**